File: week_01/twoSum.go
func twoSum(nums []int, target int) []int {
    idx := map[int]int{}  // 创建一个空的哈希表
    for j, x := range nums {  // 枚举 j
        // 在左边找 nums[i], 满足 nums[i] + x = target
        if i, ok := idx[target - x]; ok {
            return []int{i, j} // 返回两个数的下标
        }
        idx[x] = j;  // 保存 nums[j] 和 j
    }
    return nil
}

File: group_01/twoSum.cpp
```cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashtable;
        for (int i = 0; i < nums.size(); i++) {
            auto it = hashtable.find(target - nums[i]);
            if (it != hashtable.end()) {
                // it->first 是键, it->second 是值
                // (*it).second 等价于 it->second
                return {i, it->second}; 
            }
            hashtable[nums[i]] = i;
        }
        return {};
    }
};
```
